hybris/egl/ws.c

```diff
@@ -21,13 +21,23 @@
 #include <assert.h>
 #include <stdio.h>
 #include <sys/auxv.h>
+#include <pthread.h>
+#include <string.h>
+
+pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
 
 static struct ws_module *ws = NULL;
 
 static void _init_ws()
 {
 	if (ws == NULL)
 	{
+		pthread_mutex_lock(&mutex);
+		if (ws != NULL) {
```

Comment:
how about moving the pthread_mutex_(un)lock in the scope of _init_ws i.e. move it out from if (ws == NULL) to avoid that extra if (ws != NULL) check?